dart/simulation/World.cpp

```diff
@@ -74,11 +76,49 @@ World::~World()
   delete mConstraintSolver;
   delete mRecording;
 
-  for (std::vector<dynamics::Skeleton*>::const_iterator it = mSkeletons.begin();
-       it != mSkeletons.end(); ++it)
+  for(common::Connection& connection : mNameConnectionsForSkeletons)
+    connection.disconnect();
+
+  for(common::Connection& connection : mNameConnectionsForFrames)
+    connection.disconnect();
+}
+
+//==============================================================================
+WorldPtr World::clone() const
+{
+  WorldPtr worldClone(new World(mName));
+
+  worldClone->setGravity(mGravity);
+  worldClone->setTimeStep(mTimeStep);
+
+  // Clone and add each Skeleton
+  for(size_t i=0; i<mSkeletons.size(); ++i)
+  {
+    worldClone->addSkeleton(mSkeletons[i]->clone());
+  }
+
+  // Clone and add each SimpleFrame
+  for(size_t i=0; i<mFrames.size(); ++i)
+  {
+    worldClone->addFrame(mFrames[i]->clone(mFrames[i]->getParentFrame()));
+  }
+
+  // For each newly cloned SimpleFrame, try to make its parent Frame be one of
+  // the new clones if there is a match. This is meant to minimize any possible
+  // interdependencies between the kinematics of different worlds.
+  for(size_t i=0; i<worldClone->getNumFrames(); ++i)
   {
-    delete (*it);
+    dynamics::Frame* current_parent =
+        worldClone->getFrame(i)->getParentFrame();
+
+    dynamics::SimpleFramePtr parent_candidate =
+        worldClone->getFrame(current_parent->getName());
+
+    if(parent_candidate)
+      worldClone->getFrame(i)->setParentFrame(parent_candidate.get());
```

Comment:
It make sense. So the parent frame of a SimpleFrame is able to be in a different World from the World of the SimpleFrame.

Beside on this, I came up with some idea that might be useful or not. That is ***master world***. Multiple worlds can have a master world like master slide of presentation tools. The difference from cloning is all the worlds share the skeletons and simple frames in the master world rather than actually creating the instances. This might be useful if we want to simulate with different parameters while having invariant things in the master world.